control_panel_api/views.py

```diff
@@ -22,28 +28,46 @@
 from control_panel_api.serializers import (
     AppS3BucketSerializer,
     AppSerializer,
-    UserAppSerializer,
     GroupSerializer,
     S3BucketSerializer,
+    UserAppSerializer,
     UserS3BucketSerializer,
     UserSerializer,
 )
 
+logger = logging.getLogger(__name__)
+
 
 class UserViewSet(viewsets.ModelViewSet):
     queryset = User.objects.all()
     serializer_class = UserSerializer
     filter_backends = (UserFilter,)
     permission_classes = (UserPermissions,)
 
+    @transaction.atomic
     def perform_create(self, serializer):
         instance = serializer.save()
-        instance.aws_create_role()
-        instance.helm_create()
 
+        try:
+            instance.aws_create_role()
+            instance.helm_create()
+        except ClientError as e:
```

Comment:
Only problem with raising and rolling back the DB changes for all boto errors is that if we try to create a user in the CP and its role is already present it will always fail.

Ideally we'd want to ignore "this specific role already exists" exceptions and consider that as a "success".

Boto exceptions are not exactly pretty so you may need to check `e.response["Error"]["Code"]` is not a certain value.